src/cmake_generator.cpp

```diff
@@ -1030,7 +1030,11 @@ void generate_cmake(const char *path, const parser::Project *parent_project) {
             auto files = std::make_pair("FILES", inst.files);
             auto configs = std::make_pair("CONFIGURATIONS", inst.configs);
             auto destination = std::make_pair("DESTINATION", inst.destination);
-            auto component = std::make_pair("COMPONENT", inst.targets.empty() ? "" : inst.targets.front());
+            auto component_name = inst.component;
+            if (component_name.empty() && !inst.targets.empty()) {
+                component_name = inst.targets.front();
+            }
+            auto component = std::make_pair("COMPONENT", component_name);
```

Comment:
Probably the issue is that if component_name is empty it emits COMPONENT “” which causes that cmake error

Reply:
> Probably the issue is that if component_name is empty it emits COMPONENT “” which causes that cmake error

```toml
[[install]]
targets = ["MHRISE"]
destination = "MHRISE"

[[install]]
destination = "MHRISE"
files = ["dependencies/openvr/bin/win64/openvr_api.dll"]
```

Produces:
```cmake
install(
	TARGETS
		MHRISE
	DESTINATION
		MHRISE
	COMPONENT
		MHRISE
)
install(
	FILES
		"dependencies/openvr/bin/win64/openvr_api.dll"
	DESTINATION
		MHRISE
)
```

So I believe the empty `COMPONENT` case is already being handled somewhere else (haven't looked into the details). I think the issue I'm running into is that, according to https://cmake.org/cmake/help/latest/command/install.html there are different forms of the `install` function and the ones handling targets and files are just different.